software/bsg_manycore_lib/bsg_tile_config_vars.c

```diff
@@ -0,0 +1,14 @@
+// these are private variables
+// we do not make them volatile
+// so that they may be cached
+
+int __bsg_x = -1;
```

Comment:
What do you think about adding `__attribute__ ((section (".dmem")))`?

Reply:
Seems fine to me?

Reply:
Like this? 

`__attribute__ ((section (".dmem"))) int __bsg_x = -1;`

Reply:
Fixed! 

And added comments

Reply:
Hmm, with the .dmem attributes it fails regression on bsg_print_stat

Reply:
@vb000 Did you mean .sdata.dmem?

Reply:
When I remove the attributes, all tests pass. When I add them in (using .dmem) it fails.

Reply:
Without attributes: 

```
Disassembly of section .sdata.dmem:

00001060 <__bsg_tile_group_id-0x4>:
    1060:       0000                    unimp
    1062:       424c                    lw      x11,4(x12)

00001064 <__bsg_tile_group_id>:
    1064:       ffff                    0xffff
    1066:       ffff                    0xffff

00001068 <__bsg_tile_group_id_y>:
    1068:       ffff                    0xffff
    106a:       ffff                    0xffff

0000106c <__bsg_tile_group_id_x>:
    106c:       ffff                    0xffff
    106e:       ffff                    0xffff

00001070 <__bsg_grid_dim_y>:
    1070:       ffff                    0xffff
    1072:       ffff                    0xffff

00001074 <__bsg_grid_dim_x>:
    1074:       ffff                    0xffff
    1076:       ffff                    0xffff

00001078 <__bsg_grp_org_y>:
    1078:       ffff                    0xffff
    107a:       ffff                    0xffff

0000107c <__bsg_grp_org_x>:
    107c:       ffff                    0xffff
    107e:       ffff                    0xffff

00001080 <__bsg_id>:
    1080:       ffff                    0xffff
    1082:       ffff                    0xffff

00001084 <__bsg_y>:
    1084:       ffff                    0xffff
    1086:       ffff                    0xffff

00001088 <__bsg_x>:
    1088:       ffff                    0xffff
    108a:       ffff                    0xffff
```

Reply:
With attributes
```
Disassembly of section .dmem:

00001060 <__bsg_tile_group_id>:
    1060:       ffff                    0xffff
    1062:       ffff                    0xffff

00001064 <__bsg_tile_group_id_y>:
    1064:       ffff                    0xffff
    1066:       ffff                    0xffff

00001068 <__bsg_tile_group_id_x>:
    1068:       ffff                    0xffff
    106a:       ffff                    0xffff

0000106c <__bsg_grid_dim_y>:
    106c:       ffff                    0xffff
    106e:       ffff                    0xffff

00001070 <__bsg_grid_dim_x>:
    1070:       ffff                    0xffff
    1072:       ffff                    0xffff

00001074 <__bsg_grp_org_y>:
    1074:       ffff                    0xffff
    1076:       ffff                    0xffff

00001078 <__bsg_grp_org_x>:
    1078:       ffff                    0xffff
    107a:       ffff                    0xffff

0000107c <__bsg_id>:
    107c:       ffff                    0xffff
    107e:       ffff                    0xffff

00001080 <__bsg_y>:
    1080:       ffff                    0xffff
    1082:       ffff                    0xffff

00001084 <__bsg_x>:
    1084:       ffff                    0xffff
    1086:       ffff                    0xffff

Disassembly of section .sdata.dmem:

00001088 <_bsg_data_end_addr-0x4>:
    1088:       0000                    unimp
    108a:       424c                    lw      x11,4(x12)
```

Reply:
I think there's something fishy going on with `__bsg_tile_group_id` here. If you're using this as a part of `.a` file, linker script automatically puts these in one of the ".dmem" sections. So, we don't need this attribute. Is that the case?

I'll try to add these attributes in the linker script refactor. You could merge these without them...

Reply:
These are part of the .a file, I don't think I've changed anything?

Regardless, I'll leave the attributes out for now.